common/src/main/java/org/opensearch/ml/common/transport/connector/MLCreateConnectorInput.java

```diff
@@ -148,7 +148,14 @@ public static MLCreateConnectorInput parse(XContentParser parser, boolean update
                     parameters = getParameterMap(parser.map());
                     break;
                 case CONNECTOR_CREDENTIAL_FIELD:
-                    credential = parser.mapStrings();
+                    // We need to filter out any key string that is trying to imitate the subfield of the secretArn of the credential map
+                    credential = new HashMap<>();
+                    Map<String, String> credentialKeyToAdd = parser.mapStrings();
+                    for (String key : credentialKeyToAdd.keySet()) {
+                        if (!key.startsWith("secretArn.")) {
```

Comment:
if it's the connector to AWS, for example sageMager, it makes sense to not allowing using scretArn.